### PR TITLE
Bug 1693024: Add CSRF to token request endpoint [3.10]

### PR DESCRIPTION
**Pull request by stlaz**
This is a 3.11 backport of #22767 which adds CSRF validation to OAuth server token requests created from its web console.

cc @enj 

csrf check tested with `openshift start master`
Changes on top of 3.11 backport - moved `oauthserver.{Mux,Endpoints}` interfaces up a level so that they can be leveraged by all the endpoints.

### CODE REVIEW

**Comment by stlaz** [comment]
Ad test failures:
- Added disable for the `TestRegistryClientAPIv1` test (<--- @openshift/sig-developer-experience that's yet another branch you'll have to fix)
- unit tests - sigkill during build
- travis - seems to be generally broken, not just for this PR (`/home/travis/gopath/src/github.com/openshift/origin/hack/lib/constants.sh: line 129: OS_GIT_MAJOR: unbound variable`  --> maybe a change in the API)
- extended_conformance_install failed in `new_app` test because `serviceaccounts "builder" not found`? Does not sound very relevant, but I did not find a BZ referencing it as a flake

**Comment by stlaz** [comment]
Updated formatting in `tokenrequest.go` so that `go-vet` is not angry with me

File: pkg/oauthserver/server/tokenrequest/tokenrequest.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func NewTokenRequest(publicMasterURL string, osinOAuthClientGetter func() (*osin
 	return &tokenRequest{
 		publicMasterURL:       publicMasterURL,
 		osinOAuthClientGetter: osinOAuthClientGetter,
-		csrf:                  csrf,
+		csrf: csrf,
```
Comment (enj) [inline pkg/oauthserver/server/tokenrequest/tokenrequest.go]: nit: should be part of 3dabe0d10458c5a36c100f6b6b18797e61dc98f5

**Comment by stlaz** [comment]
Fixed the comment + added a commit from master which should fix the integration tests.

**Pull request closed without merging**